elastic/kibana
# [Fleet] Remove non relevant field when changing output type

## Summary

Related to #https://github.com/elastic/fleet-server/issues/1256

When changing an output type from logstash to elasticsearch the SSL fields where not removed from the output, creating an invalid output.

that PR fix that

## Tests

1. You can create a logstash output output-1
2. Check in a policy using that output that you have the `ssl` property
3. update `output-1` to be an `elasticsearch` output
4. Check in a policy using that output that you do not have the `ssl` property

That change is covered by unit tests
